client/gutenberg/editor/calypsoify-iframe.tsx

```diff
@@ -116,13 +117,57 @@ class CalypsoifyIframe extends Component< Props & ConnectedProps & ProtectedForm
 	componentDidMount() {
 		MediaStore.on( 'change', this.updateImageBlocks );
 		window.addEventListener( 'message', this.onMessage, false );
+
+		this.handleThirdPartyiFrameDomains();
```

Comment:
Checking the iframe URL in `componentDidMount` is somewhat late: at that time, the `iframe` is already rendered and its load request is being issued.

It's hard to find a good place for this check (component constructor is much better, but still not ideal). I would personally do it in a page.js handler in `gutenberg/editor/controller`.

Reply:
have moved to the controller, that did seem like a nice logical home for this.